File: sorting_algorithms/quick_sort.py
def quick_sort(nums, low, high):
    if low < high:
        middle = partition(nums,low,high)
        quick_sort(nums, low, middle - 1)
        quick_sort(nums,middle + 1, high)

def partition(nums, low, high):
    pivot = nums[high]
    i = low
    for j in range(low, high):
        if nums[j] < pivot :
            nums[i], nums[j] = nums[j], nums[i]
            i += 1
    nums[i],nums[high] = nums[high],nums[i]
    return i

arr = [2,1,3]  
high = len(arr) - 1
quick_sort(arr, 0, high)
print(f"Sorted array: {arr}")

File: quick_sort.go
package main

func quickSort(arr []int)[]int {
	if len(arr) < 2 {
		return arr
	}

	pivot := arr[0]
	less := make([]int, 0)
	greater := make([]int, 0)

	for _, val := range arr[1:] {
		if val > pivot {
			greater = append(greater, val)
		}else {
			less = append(less, val)
		}
	}

	result := quickSort(less)
	result = append(result, pivot)
	result = append(result, quickSort(greater)...)
	return result
}